### PR TITLE
chore: `settings/developer` - remove pages router and use app router

### PR DESCRIPTION
**Pull request by hbjORbj**
## What does this PR do?

- Fixes CAL-4404
- remove pages router and use app router for all pages in `settings/developer`
- remove env vars not needed anymore as a result

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A - I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

- all settings/developer/**/* pages must work as expected

### CODE REVIEW

File: apps/web/playwright/webhook.e2e.ts [hunk 1]
```diff
@@ -814,7 +813,7 @@ test.describe("OOO_CREATED", async () => {
       hasTeam: true,
     });
     await user.apiLogin();
-    const { webhookReceiver, teamId } = await webhooks.createTeamReceiver();
```
Comment (hbjORbj) [inline apps/web/playwright/webhook.e2e.ts:817]: teamId not used

File: apps/web/playwright/webhook.e2e.ts [hunk 2]
```diff
@@ -490,7 +490,6 @@ test.describe("MEETING_ENDED, MEETING_STARTED", async () => {
   }, _testInfo) => {
     const user = await users.create();
     await user.apiLogin();
-    const tomorrow = dayjs().add(1, "day");
```
Comment (hbjORbj) [inline apps/web/playwright/webhook.e2e.ts:493]: tomorrow not used

**Review by joeauyeung — DISMISSED** [review]

File: apps/web/playwright/fixtures/webhooks.ts [hunk 3]
```diff
@@ -12,7 +12,6 @@ export function createWebhookPageFixture(page: Page) {
       await page.waitForURL((u) => u.pathname === "/settings/developer/webhooks/new");
       const url = page.url();
       const teamId = Number(new URL(url).searchParams.get("teamId")) as number;
-      await page.click('[data-testid="new_webhook"]');
```
Comment (hbjORbj) [inline apps/web/playwright/fixtures/webhooks.ts:15]: Hmm this is wrong. I don't know how this e2e test passed. `/settings/developer/webhooks/new` is expected to not have this button.

**Comment by hbjORbj** [comment]
TODO: Fix "Create Webhook" Button in settings/developer/webhooks/new

**Review by joeauyeung — APPROVED** [review]

**Review by joeauyeung — APPROVED** [review]

**Pull request merged**